public/app/core/utils/explore.ts

```diff
@@ -358,35 +398,29 @@ export function getIntervals(range: TimeRange, lowLimit: string, resolution: num
   return kbn.calculateInterval(range, resolution, lowLimit);
 }
 
-export const makeTimeSeriesList: ResultGetter = (dataList, transaction, allTransactions) => {
+export const makeTimeSeriesList = (seriesData: DataQueryResponseData[]) => {
+  const result: TimeSeries[] = [];
```

Comment:
I have not dug into this too deeply, but this looks like it would not safely handle `TableData|TimeSeries|SeriesData`,  I expect you will need something like:

https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/graph/module.ts#L194
and
https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/graph/data_processor.ts#L14

To safely convert any input into `app/core/time_series2` 

to convert any response to TimeSeries[]

Reply:
Thanks, will look into that!

Reply:
Does the new implementation of `makeTimeSeriesList` feel safer @ryantxu ?

Reply:
much better